students/DrewSmith/lessons/lesson05/assignment/database.py

```diff
@@ -0,0 +1,133 @@
+        self.port = port
+        self.connection = None
+
+    def __enter__(self):
+        self.connection = MongoClient(self.host, self.port)
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.connection.close()
+
+def import_data(directory_name, product_file, customer_file, rentals_file,
+                connection=None, database_name=None):
+    '''
+    Imports data from 3 files an inserts product, customer and rental data
+
+    :param directory_name: directory where data files are stored
+    :param product_file: file name with product data
+    :param customer_file: file name with customer data
+    :param rentals_file: file name with rentals data
+    :param connection: DB connection for data inserts
+    :param database_name: Database to use for the query,
+        or None to use production default
+    '''
+    connection = connection or MongoDBConnection()
+    database_name = database_name or config.DATABASE_NAME
+    directory = pathlib.Path(directory_name)
+
+    errors = [0, 0, 0]
+    counts = [0, 0, 0]
+    with connection:
+        database = connection.connection.get_database(name=database_name)
+
+        with open(directory / product_file, 'r') as file:
+            csv_file = csv.DictReader(file, fieldnames=("product_id",
+                                                        "description",
+                                                        "product_type",
+                                                        "quantity_available"))
+            records = list()
+            for index, record in enumerate(csv_file):
+                if index == 0:
+                    continue
+                try:
+                    record["quantity_available"] = int(record["quantity_available"])
+                except ValueError:
+                    errors[0] += 1
+                else:
+                    records.append(record)
+            counts[0] = len(records)
+            database["products"].insert_many(records)
+
+        with open(directory / customer_file, 'r') as file:
+            csv_file = csv.DictReader(file, fieldnames=("user_id",
+                                                        "name",
+                                                        "address",
+                                                        "zip_code",
+                                                        "phone_number",
+                                                        "email"))
+            records = [record for index, record in enumerate(csv_file) if index > 0]
+            counts[1] = len(records)
+            database["customers"].insert_many(records)
+
+        with open(directory / rentals_file, 'r') as file:
+            csv_file = csv.DictReader(file, fieldnames=("product_id", "user_id"))
+            records = [record for index, record in enumerate(csv_file) if index > 0]
+            counts[2] = len(records)
+            database["rentals"].insert_many(records)
```

Comment:
Nice comprehensions. :)
This is a lot for one function, I would think about breaking some of this out as separate functions. Each of your context manager blocks look pretty similar, I bet you could take advantage of that. Also it seems you only expect to get one type of error, and only for one file. That seems optimistic. ;-)